### DISCUSSION TITLE
How to install package to specific directories?

### DISCUSSION THREAD

**Discussion by nekomiya-kasane** [xmake-io/xmake#3779] [Q&A]
I used `add_require` to fetch third-party libraries. XMake fetch them to `C:\Users\username\AppData\Local\.xmake\packages`, how can I redefine this path so as to make it installed into `/root/to/my/project/third_party` (functions only for the current project, not globally)?

**Comment by waruqi** [comment]
https://xmake.io/#/guide/configuration?id=xmake_pkg_installdir

or enable require-lock, it will also install local build directory. https://xmake.io/#/package/remote_package?id=dependent-package-lock-and-upgrade

**Reply by nekomiya-kasane** [reply]
Could you please provide some details? I tried 
```
on_load(function (package) 
    if not has_config("pkg_installdir") then
        local third_party_dir = "$(projectdir)/ThirdParty"
        os.mkdir(third_party_dir)
        set_config("pkg_installdir", third_party_dir)
    end
    printf("Third party library directory: " .. get_config("pkg_installdir") .. "\n")
end)
```
But xmake still uses packages downloaded to the previous directory:
```
2a631882d
checking for cmake ... no
checking for cmake ... D:\scoop_global\apps\mingw-winlibs\current\bin\cmake
checking for xmake::jsoncpp ... no
checking for jsoncpp ... vcpkg::jsoncpp 1.9.5
checking for xmake::eigen ... eigen 3.4.0
```
I want to force xmake to download all required packages (`add_require(...)`) to a local folder and use their local copies.

Should I exec a shell command (`set XMAKE_PKG_INSTALLDIRid=...`) in `xmake.lua`? But this seems incorrect.

**Reply by waruqi** [reply]
> Should I exec a shell command (set XMAKE_PKG_INSTALLDIRid=...) in xmake.lua? But this seems incorrect.

XMAKE_PKG_INSTALLDIRid -> XMAKE_PKG_INSTALLDIR
